src/boot/efi/boot.c

```diff
@@ -1098,10 +1195,82 @@ static void config_entry_free(ConfigEntry *entry) {
         free(entry);
 }
 
+static void menu_free(Menu *menu);
+
+static void entry_free(Entry *entry) {
+        if (!entry)
+                return;
+
+        if (entry->entry_type == CONFIG_ENTRY) {
+                config_entry_free(entry->entry_data.config_entry);
+        } else if (entry->entry_type == MENU) {
+                menu_free(entry->entry_data.menu);
+        }
+
+        // Finally, free the entry itself
```

Comment:
coding style. we use /* */ for explanatory comments, and // for fixme comments only.